### PR TITLE
Qa canonizer sai

### PR DESCRIPTION
**Pull request by saideekshith8**
modified test cases.

### CODE REVIEW

File: CanonizerAcoountSettingPage.py [hunk 1]
```diff
@@ -0,0 +1,432 @@
+from time import sleep
+
+# import page as page
+from selenium.webdriver.common.keys import Keys
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:4]: Looks like it is unused. Please remove if it is not used anywhere

File: CanonizerAcoountSettingPage.py [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+from selenium.common.exceptions import NoSuchElementException
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:5]: Looks like it is unused. Please remove if it is not used anywhere

File: CanonizerAcoountSettingPage.py [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+import datetime
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:7]: Looks like it is unused. Please remove if it is not used anywhere

File: CanonizerAcoountSettingPage.py [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*CanonizerSupportCampIdentifiersPage.DIRECET_SUPPORTED_CAMPS).click()
+        self.find_element(*CanonizerSupportCampIdentifiersPage.CLICKING_ON_TOPIC_NAME).CLICK()
+        title = self.find_element(*CanonizerSupportCampIdentifiersPage.TOPIC_NAME_TITLE).text
+        if title == 'Topic: automation 123':
+            return CanonizerSupportCamps(self.driver)
+        else:
+            print("title not found")
+
+    def verify_Remove_Support_button_functionality(self):
+        self.hover(*CanonizerSupportCampIdentifiersPage.CLICK_ON_DROPDOWN)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:417]: Duplicate piece of code from line 32. Please remove the duplicates

File: CanonizerAcoountSettingPage.py [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+
+    def verifying_supported_camps(self):
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.SUPPORTED_CAMPS)
+        self.find_element(*AccountSettingPageIdentifier.SUPPORTED_CAMPS).click()
+        title = self.find_element(*AccountSettingPageIdentifier.SUPPORTED_CAMPS).text
+        if title == 'Supported Camps':
+            return CanonizerAccountSettingPage(self.driver)
+
+    def check_the_validation_for_Phone_Number_field(self, number):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:74]: Incorrect function name.

File: CanonizerAcoountSettingPage.py [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.PROFILE_BUTTON)
+        self.find_element(*AccountSettingPageIdentifier.PROFILE_BUTTON).click()
+        self.hover(*AccountSettingPageIdentifier.PHONE_NUMBER)
+        self.find_element(*AccountSettingPageIdentifier.PHONE_NUMBER).send_keys(number)
+        title = self.find_element(*AccountSettingPageIdentifier.PHONE_NUMBER_ERROR).text
+        if title == 'Phone number must be at least 10 digits!':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:81]: Must be come from a common message file

File: CanonizerAcoountSettingPage.py [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+            return CanonizerAccountSettingPage(self.driver)
+
+    def check_the_Functionality_of_verify_button_without_entering_the_data(self):
+        self.click_account_settings_page_button()
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:85]: This code is repeated in the Next subroutine. Needs to refactor

File: CanonizerAcoountSettingPage.py [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+    def verify_the_functionality_of_mobile_carrier_drop_down(self):
+        self.click_account_settings_page_button()
+        self.find_element(*AccountSettingPageIdentifier.PROFILE_BUTTON).click()
+        self.hover(*AccountSettingPageIdentifier.MOBILE_CARRIER)
+        self.hover(*AccountSettingPageIdentifier.MOBILE_CARRIER)
+        self.find_element(*AccountSettingPageIdentifier.MOBILE_CARRIER).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_all_the_fields_are_present_in_Personal_Information_field(self):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:102]: Function name should be lowercase

File: CanonizerAcoountSettingPage.py [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.FIRST_NAME)
+        self.hover(*AccountSettingPageIdentifier.MIDDLE_NAME)
+        self.hover(*AccountSettingPageIdentifier.LAST_NAME)
+        self.hover(*AccountSettingPageIdentifier.GENDER)
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_the_validation_for_first_name(self, first):
+        self.click_account_settings_page_button()
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:112]: Code repeated in the next function

File: CanonizerAcoountSettingPage.py [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+
+    def verify_the_functionality_of_radio_button_in_selecting_the_gender(self):
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.GENDER)
+        self.find_element(*AccountSettingPageIdentifier.GENDER).click()
+        self.find_element(*AccountSettingPageIdentifier.UPDATE_BUTTON).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_the_functionality_of_selecting_the_DOB(self, dob):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:137]: Function name should be lowercase

File: CanonizerAcoountSettingPage.py [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+        sleep(3)
+        self.find_element(*AccountSettingPageIdentifier.DATA_BIRTH).click()
+        sleep(3)
+        self.find_element(*AccountSettingPageIdentifier.DATA_BIRTH).clear()
+        self.find_element(*AccountSettingPageIdentifier.DATA_BIRTH).send_keys(dob)
+        self.find_element(*AccountSettingPageIdentifier.UPDATE_BUTTON).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_when_user_updated_the_Personal_Information_and_logs_out_and_logIn_again(self, first, middle, last):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:148]: Function name should be lowercase

File: CanonizerAcoountSettingPage.py [hunk 12]
```diff
@@ -0,0 +1,432 @@
...
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.FIRST_NAME)
+        self.find_element(*AccountSettingPageIdentifier.FIRST_NAME).send_keys(first)
+        self.find_element(*AccountSettingPageIdentifier.MIDDLE_NAME).send_keys(middle)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:152]: Code repeated in the next function. Refactor required

File: CanonizerAcoountSettingPage.py [hunk 13]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*AccountSettingPageIdentifier.LAST_NAME).send_keys(last)
+        self.find_element(*AccountSettingPageIdentifier.GENDER).click()
+        self.find_element(*AccountSettingPageIdentifier.UPDATE_BUTTON).click()
+        self.find_element(*AccountSettingPageIdentifier.CLICK_ON_DROPDOWN).click()
+        self.find_element(*AccountSettingPageIdentifier.LOGOUT).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_when_user_click_on_change_password_tab_its_navigating_to_change_Password_Page(self):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:175]: Function name should be lowercase

File: CanonizerAcoountSettingPage.py [hunk 14]
```diff
@@ -0,0 +1,432 @@
...
+        self.click_account_settings_page_button()
+        sleep(5)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:177]: Not required

File: CanonizerAcoountSettingPage.py [hunk 15]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        self.hover(*AccountSettingPageIdentifier.CURRENT_PASSWORD)
+        self.hover(*AccountSettingPageIdentifier.NEW_PASSWORD)
+        self.hover(*AccountSettingPageIdentifier.CONFIRM_PASSWORD)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_keeping_all_the_fields_empty_and_click_on_save(self, current, new, confirm):
+        self.click_account_settings_page_button()
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:195]: Repeated code. Please remove the duplicate code.

File: CanonizerAcoountSettingPage.py [hunk 16]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*AccountSettingPageIdentifier.CURRENT_PASSWORD).send_keys(current)
+        self.find_element(*AccountSettingPageIdentifier.NEW_PASSWORD).send_keys(new)
+        self.find_element(*AccountSettingPageIdentifier.CONFIRM_PASSWORD).send_keys(confirm)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+
+        return CanonizerAccountSettingPage(self.driver)
+
+    def verify_entering_the_invalid_current_password(self, current):
+        self.click_account_settings_page_button()
+        sleep(5)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:207]: Sleep is unnecessary.

File: CanonizerAcoountSettingPage.py [hunk 17]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        self.find_element(*AccountSettingPageIdentifier.CURRENT_PASSWORD).send_keys(current)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+        title = self.find_element(*AccountSettingPageIdentifier.CURRENT_PASSWORD_ERROR).text
+        if title == 'Incorrect Current Password':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:212]: All the messages comes from the common place. Please create a separate file or Class and use it for the messages

File: CanonizerAcoountSettingPage.py [hunk 18]
```diff
@@ -0,0 +1,432 @@
...
+            return CanonizerAccountSettingPage(self.driver)
+
+    def verify_entering_the_invalid_new_password(self, new):
+        self.click_account_settings_page_button()
+        sleep(5)
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:217]: Please remove sleep

File: CanonizerAcoountSettingPage.py [hunk 19]
```diff
@@ -0,0 +1,432 @@
...
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        self.find_element(*AccountSettingPageIdentifier.NEW_PASSWORD).send_keys(new)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+        title = self.find_element(*AccountSettingPageIdentifier.NEW_PASSWORD_ERROR).text
+        if title == 'Password must be contain small, capital letter, number and special character like Abc@1234.':
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:222]: All the messages must comes from the common file

File: CanonizerAcoountSettingPage.py [hunk 20]
```diff
@@ -0,0 +1,432 @@
...
+        self.click_account_settings_page_button()
+        self.hover(*AccountSettingPageIdentifier.CHANGE_PASSWORD)
+        self.find_element(*AccountSettingPageIdentifier.CHANGE_PASSWORD).click()
+        self.find_element(*AccountSettingPageIdentifier.CONFIRM_PASSWORD).send_keys(DEFAULT_INVALID_CONFIRM_PASSWORD)
+        self.find_element(*AccountSettingPageIdentifier.SAVE_BUTTON).click()
+        title = self.find_element(*AccountSettingPageIdentifier.CONFIRM_PASSWORD_ERROR).text
+        if title == 'Confirm Password does not match.':
+            return CanonizerAccountSettingPage(self.driver)
+
+    def verify_when_both_new_Password_and_confirm_Password_does_not_match(self, DEFAULT_NEW_PASSWORD, DEFAULT_CONFIRM_PASSWORD):
```
Comment (akuks) [inline CanonizerAcoountSettingPage.py:235]: Function name should be lowercase

**Review by akuks — CHANGES_REQUESTED** [review]
Please do the requested changes. There are many errors in the file. so I haven't gone pass to the first file

**Comment by saideekshith8** [comment]
Need to modified

**Pull request closed without merging**